src/compiler/parser.ts

```diff
@@ -6259,7 +6259,7 @@ namespace ts {
                 }
 
                 function parseJSDocIdentifier(): Identifier {
-                    if (token !== SyntaxKind.Identifier) {
+                    if (token !== SyntaxKind.Identifier && !isTSOnlyKeyword(token)) {

```

Comment:
`parseJSDocIdentifier` doesn't work inside a type expression. The difference cases should be:

```js
/** param {myNonExistingTypeName} x **/ // ok
/** param {myNonExistingTypeName} import **/ // not ok
/** param {myNonExistingTypeName} type **/ // should be ok, but not now
```